### ISSUE TITLE
How do you calculate the average precision in your paper

### ISSUE THREAD

**Issue by LZDSJTU** [DavyNeven/fastSceneUnderstanding#12]
Your network has two outputs: semantic predictions and embeddings. Then you get instances by clustering embeddings. 

However, it seems that you do not get a score for each instance. So how do you get the PR curve and then calculate the AP metric?

**Comment by davyneven** [comment]
Indeed, our method does not output a score for each mask, so all masks get a score of 1. For evaluating and calculating the AP metric, we use the official Cityscapes evaluation scripts, which you can find here: https://github.com/mcordts/cityscapesScripts

**Comment by LZDSJTU** [comment]
Hello~

I have successfully trained the semantic segmentation with a discriminative loss function. However, I find that the discriminative loss cannot decrease easily.

I want to ask your tricks when training.

I do the following steps:

1. I train a semantic segmentation network.
2. I choose a pre-trained model from semantic segmentation network to train the instance model
(Question#1: I do not exactly know which pre-trained model is better (For example, model with 50000 steps or 100000 steps)
3. I choose the margin value of 0.5 and 1.5 for var and dist
4. I use a learning rate of 0.0005

I think that many factors will influence the final training results. So I want to ask your experience when training.

Thank you very much~



lzd950512@sjtu.edu.cn
 
From: DavyNeven
Date: 2018-11-26 20:57
To: DavyNeven/fastSceneUnderstanding
CC: LZDSJTU; Author
Subject: Re: [DavyNeven/fastSceneUnderstanding] How do you calculate the average precision in your paper (#12)
Indeed, our method does not output a score for each mask, so all masks get a score of 1. For evaluating and calculating the AP metric, we use the official Cityscapes evaluation scripts, which you can find here: https://github.com/mcordts/cityscapesScripts
―
